core/src/impl/Kokkos_Core.cpp

```diff
@@ -282,11 +295,206 @@ void initialize_profiling(const InitArguments& args) {
     std::exit(EXIT_SUCCESS);
   }
   Kokkos::Tools::parseArgs(args.tool_args);
+  for (const auto& category_values : Kokkos::Impl::metadata_map) {
+    for (const auto& key_value : category_values.second) {
+      Kokkos::Tools::declareMetadata(key_value.first, key_value.second);
```

Comment:
So you just throw away the "category" info?

Reply:
In declaring metadata to tools, yes. The category exists only to make metadata printing behave the same way